handlers/tokeninfo/proxy/handler.go

```diff
@@ -22,7 +24,7 @@ type tokenInfoProxyHandler struct {
 
 // NewTokenInfoProxyHandler returns an http.Handler that proxies every Request to the server
 // at the upstreamURL
-func NewTokenInfoProxyHandler(upstreamURL *url.URL, cacheMaxSize int64, cacheTTL time.Duration) http.Handler {
+func NewTokenInfoProxyHandler(upstreamURL *url.URL, cacheMaxSize int64, cacheTTL time.Duration) tokeninfo.Handler {
```

Comment:
why the tokeninfo.Handler ? If it needs to be like that, then the godoc needs to be updated accordingly